lib/node_modules/@stdlib/repl/lib/eager_evaluator.js

```diff
@@ -0,0 +1,326 @@
+
+'use strict';
+
+// MODULES //
+
+var readline = require( 'readline' );
+var inspect = require( 'util' ).inspect;
+var logger = require( 'debug' );
+var parse = require( 'acorn' ).parse;
+var replace = require( '@stdlib/string/replace' );
+var setNonEnumerableReadOnly = require( '@stdlib/utils/define-nonenumerable-read-only-property' );
+var copy = require( '@stdlib/array/base/copy' );
+var max = require( '@stdlib/math/base/special/max' );
+var processCommand = require( './process_command.js' );
+var compileCommand = require( './compile_command.js' );
+var ANSI_COLORS = require( './ansi_colors.js' );
+
+
+// VARIABLES //
+
+var debug = logger( 'repl:eager-evaluator' );
+var AOPTS = {
+	'ecmaVersion': 'latest'
+};
+var ROPTS = {
+	'timeout': 100, // (in milliseconds) this controls how long eagerly evaluated commands have to execute; we need to avoid setting this too high in order to avoid eager evaluation interfering with the UX when naturally typing
+	'displayErrors': false,
+	'breakOnSigint': true // Node.js >=6.3.0
+};
+var tempDB = {
+	'base_sin': {
+		'isPure': true
+	}
+};
+var ANSI_GRAY = ANSI_COLORS[ 'brightBlack' ];
+var ANSI_RESET = ANSI_COLORS[ 'reset' ];
+
+
+// FUNCTIONS //
+
+/**
+* Recursively traverses the node to determine whether the node is side-effect-free.
+*
+* @private
+* @param {Object} node - ast node
+* @returns {boolean} boolean indicating whether the node is side-effect-free
+*/
+function traverse( node ) {
+	var fname;
+	var i;
+	if ( !node ) {
+		return false;
+	}
+	if ( node.type === 'Literal' || node.type === 'Identifier' || node.type === 'MemberExpression' ) {
+		return true;
+	}
+	if ( node.type === 'BinaryExpression' ) {
+		if ( traverse( node.left ) && traverse( node.right ) ) {
+			return true;
+		}
+	} else if ( node.type === 'ExpressionStatement' ) {
+		if ( traverse( node.expression ) ) {
+			return true;
+		}
+	} else if ( node.type === 'CallExpression' ) {
+		fname = getFunctionName( node.callee );
+		if ( tempDB[fname] && tempDB[fname].isPure ) {
+			// Examine each function argument for potential side-effects...
+			for ( i = 0; i < node.arguments.length; i++ ) {
+				if ( !traverse( node.arguments[ i ] ) ) {
+					return false;
+				}
+			}
+			return true;
+		}
+	}
+	return false;
+}
+
+/**
+* Resolves the function name associated with a provided AST node.
+*
+* @private
+* @param {Object} node - ast node
+* @returns {string} function name representing the node
+*/
+function getFunctionName( node ) {
+	if ( !node ) {
+		return '';
+	}
+	if ( node.type === 'MemberExpression' ) {
+		return getFunctionName( node.object ) + '_' + node.property.name;
+	}
+	if ( node.type === 'Identifier' ) {
+		return node.name;
+	}
+	return '';
+}
+
+
+// MAIN //
+
+/**
+* Constructor for creating an eager evaluator.
+*
+* @private
+* @param {REPL} repl - repl instance
+* @param {Object} rli - readline instance
+* @param {Boolean} enabled - boolean indicating whether the eager evaluator should be initially enabled
```

Comment:
```suggestion
* @param {boolean} enabled - boolean indicating whether the eager evaluator should be initially enabled
```

`Boolean` indicates a class instance. That is not what is meant here. `boolean` means a boolean primitive.